include/forcing/DataProviderSelectors.hpp

```diff
@@ -0,0 +1,178 @@
+#ifndef NGEN_DATA_SELECTORS_HPP
+#define NGEN_DATA_SELECTORS_HPP
+
+#include <string>
+
+/**  
+ * @brief This class is intended to be the virtual base for all data selectors used with the 
+ * DataProvider API
+ * 
+ */
+
+class CatchmentAggrDataSelector
```

Comment:
This might be worth renaming in the future if we have CatchmentAggregate in the data model but that's a relatively easy refactor later.